src/app/node-data/component.ts

```diff
@@ -156,9 +162,44 @@ export class NodeDataComponent extends BaseFormValidator implements OnInit, OnDe
         }
       });
 
+    // Cross validate autoscaling options.
+    merge(
+      this.form.get(Controls.MinReplicas).valueChanges,
+      this.form.get(Controls.MaxReplicas).valueChanges,
+      this.form.get(Controls.UseAutoscaling).valueChanges
+    )
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(_ => {
+        const useAutoscaling = !!this.form.get(Controls.UseAutoscaling).value;
+        if (!useAutoscaling) {
+          this.form.get(Controls.MinReplicas).setErrors(null);
+          this.form.get(Controls.MaxReplicas).setErrors(null);
+          return;
+        }
+
+        const minR = this.form.get(Controls.MinReplicas).value || 0;
+        if (minR < 0) {
+          this.form.get(Controls.MinReplicas).setErrors({min: true});
```

Comment:
It's in the template. You can also use ` Validators.min(0)`.

Reply:
In general I think you can set validators for both fields and then keep them defaulted like now. If user will not select the autoscaling checkbox then the form will still be valid. You can also make them disabled/enabled depending on if the checkbox is marked or not.